plugin/audit_log/audit_log.c

```diff
@@ -493,34 +543,34 @@ size_t audit_log_connection_record(char *buf, size_t buflen,
                      "\"%s\",\"%s\",\"%s\",\"%lu\",%d,\"%s\",\"%s\",\"%s\","
                      "\"%s\",\"%s\",\"%s\",\"%s\"\n" };
 
-  return my_snprintf(buf, buflen,
-                     format_string[audit_log_format],
-                     name,
-                     make_record_id(id_str, sizeof(id_str)),
-                     make_timestamp(timestamp, sizeof(timestamp), t),
-                     event->thread_id,
-                     event->status,
-                     escape_string(event->user,
-                                   event->user_length,
-                                   endptr, endtmp - endptr, &endptr),
-                     escape_string(event->priv_user,
-                                   event->priv_user_length,
-                                   endptr, endtmp - endptr, &endptr),
-                     escape_string(event->external_user,
-                                   event->external_user_length,
-                                   endptr, endtmp - endptr, &endptr),
-                     escape_string(event->proxy_user,
-                                   event->proxy_user_length,
-                                   endptr, endtmp - endptr, &endptr),
-                     escape_string(event->host,
-                                   event->host_length,
-                                   endptr, endtmp - endptr, &endptr),
-                     escape_string(event->ip,
-                                   event->ip_length,
-                                   endptr, endtmp - endptr, &endptr),
-                     escape_string(event->database,
-                                   event->database_length,
-                                   endptr, endtmp - endptr, &endptr));
+  user= escape_string(event->user, event->user_length,
+                      endptr, endbuf - endptr, &endptr, NULL);
+  priv_user= escape_string(event->priv_user,
+                           event->priv_user_length,
+                           endptr, endbuf - endptr, &endptr, NULL);
+  external_user= escape_string(event->external_user,
+                               event->external_user_length,
+                               endptr, endbuf - endptr, &endptr, NULL);
+  proxy_user= escape_string(event->proxy_user, event->proxy_user_length,
+                            endptr, endbuf - endptr, &endptr, NULL);
+  host= escape_string(event->host, event->host_length,
+                      endptr, endbuf - endptr, &endptr, NULL);
+  ip= escape_string(event->ip, event->ip_length,
+                    endptr, endbuf - endptr, &endptr, NULL);
+  database= escape_string(event->database, event->database_length,
+                          endptr, endbuf - endptr, &endptr, NULL);
+
+  len= my_snprintf(endptr, endbuf - endptr,
```

Comment:
Yes. It is should be enough. USER, IP, HOST, DATABASE all have strictly limited length.